app/services/order_tax_adjustments_fetcher.rb

```diff
@@ -10,25 +9,13 @@ def initialize(order)
 
   def totals
     order.all_adjustments.tax.each_with_object({}) do |adjustment, hash|
-      tax_rates_hash = tax_rates_hash(adjustment)
-      hash.update(tax_rates_hash) { |_tax_rate, amount1, amount2| amount1 + amount2 }
+      tax_rate = adjustment.originator
+      tax_amounts = { tax_rate => adjustment.amount }
+      hash.update(tax_amounts) { |_tax_rate, amount1, amount2| amount1 + amount2 }
```

Comment:
Wow, so before, each adjustment could have multiple tax rates but now it's only one, right?

We could probably simplify this further:
```suggestion
      tax_rate = adjustment.originator
      hash[tax_rate] = hash[tax_rate].to_f + adjustment.amount
```

Reply:
> Wow, so before, each adjustment could have multiple tax rates but now it's only one, right?

Multiple tax adjustments for different rates can still be added to a single item! :heavy_check_mark: 

In this case though, we're looking at tax adjustments that have _already been created_ (`order.all_adjustments.tax`) and we're summing them by rate.

The complexity in this class was related to the previous use of the `included_tax` attribute, where theoretically we could have been squashing multiple tax amounts from multiple rates into a single database column (with no reference to the rates). Figuring out what rates were used in that case meant we had to attempt to reverse-engineer it to see if multiple rates had been used, and then try to compute both. Now that all tax amounts are stored in _distinct adjustments_ (which each clearly reference the rate that was used to create them), that problem totally disappears.

Does that make sense? :sweat_smile:

Reply:
Yes. And it makes me really appreciate the great work you've done to clean up this mess. :medal_sports: